pennylane/resource/error/error.py

```diff
@@ -142,3 +143,25 @@ def get_error(approximate_op: Operator, exact_op: Operator):
         m1 = qml.matrix(exact_op, wire_order=wire_order)
         m2 = qml.matrix(approximate_op, wire_order=wire_order)
         return qml.math.max(qml.math.svd(m1 - m2, compute_uv=False))
+
+
+def _compute_algo_error(tape) -> Dict[str, AlgorithmicError]:
+    """Given a quantum circuit (tape), this function computes the algorithmic error
+    generated by standard PennyLane operations.
+
+    Args:
+        tape (.QuantumTape): The quantum circuit for which we compute errors
+
+    Returns:
+        dict[str->.AlgorithmicError]: dict with error name and combined error as key-value pair.
+    """
+
+    op_errors = {}
+    for op in tape.operations:
+        if issubclass(op.__class__, ErrorOperation):
+            error_val = op.error()  # get error object from the op
+            error_str = getattr(error_val.__class__, "__name__")  # get name of the class as str
+            error_obj = op_errors.get(error_str, None)  # attempt obtaining error from dict
+            op_errors[error_str] = error_val if error_obj is None else error_obj.combine(error_val)
```

Comment:
Initializes the dictionary key if the error `type` is not already present; otherwise, would combine it with the next encountered error of the same `type`.